hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/common/AutoCloseDataSetLock.java

```diff
@@ -30,17 +33,40 @@
  * use a try-with-resource syntax.
  */
 public class AutoCloseDataSetLock extends AutoCloseableLock {
-  private Lock lock;
+
+  private final Lock lock;
+
+  private final DataNodeLockManager<AutoCloseDataSetLock> dataNodeLockManager;
+
+  private final boolean isReadLock;
```

Comment:
Not sure if we need this attribute (of course include this attribute at `LockHeldInfo` and `DataSetLockHeldInfo`)

Reply:
@Hexiaoqiao Thank you for your feedback. My initial thought was to distinguish between read and write lock because they have different impacts on concurrency. Therefore, I added this attribute here and passed it along to the subsequent lock hold information.